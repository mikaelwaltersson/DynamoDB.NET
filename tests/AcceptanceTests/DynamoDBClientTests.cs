
using Amazon.DynamoDBv2;
using DynamoDB.Net.Model;
using DynamoDB.Net.Serialization;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;

namespace DynamoDB.Net.Tests.AcceptanceTests;

public partial class DynamoDBClientTests : IAsyncLifetime
{
    IServiceProvider serviceProvider = null!;
    IAmazonDynamoDB dynamoDB = null!;
    IDynamoDBClient dynamoDBClient = null!;
    string tableName = null!;

    async Task IAsyncLifetime.InitializeAsync()
    {
        var services = new ServiceCollection();

        services.AddOptions();
        services.AddLogging();
        
        services.AddSingleton(
            typeof(IAmazonDynamoDB),
            serviceProvider =>
                new AmazonDynamoDBClient(
                    new AmazonDynamoDBConfig
                    {
                        ServiceURL = "http://localhost:8000"
                    }));

        services.AddSingleton<IDynamoDBClient, DynamoDBClient>();
        services.Configure<DynamoDBClientOptions>(
            options => 
            {
                options.TableNamePrefix = $"acceptance-tests-{Guid.NewGuid()}";
            }
        );

        serviceProvider = services.BuildServiceProvider();

        dynamoDB = serviceProvider.GetRequiredService<IAmazonDynamoDB>();
        
        var createTableRequest = 
            TableDescription.Get(
                typeof(TestModels.UserPost), 
                JsonContractResolver.DefaultDynamoDB).
                    GetCreateTableRequest(
                        serviceProvider.GetRequiredService<IOptions<DynamoDBClientOptions>>().Value);

        await dynamoDB.CreateTableAsync(createTableRequest);

        dynamoDBClient = serviceProvider.GetRequiredService<IDynamoDBClient>();
        tableName = createTableRequest.TableName;
    }

    async Task IAsyncLifetime.DisposeAsync()
    {
        var dynamoDb = serviceProvider.GetRequiredService<IAmazonDynamoDB>();
        
        await dynamoDb.DeleteTableAsync(tableName);
    }
}
