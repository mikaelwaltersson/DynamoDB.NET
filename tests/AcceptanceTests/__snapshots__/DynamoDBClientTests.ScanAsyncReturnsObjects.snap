[
  {
    "UserId": "51b78930-5854-4e91-b4c2-000000000002",
    "Timestamp": "2022-10-18T16:02:00Z",
    "RoleIds": [
      "0ae4be37-7156-48a6-8598-000000000002",
      "9f825216-3ed1-4543-89d6-000000000002"
    ]
  },
  {
    "UserId": "51b78930-5854-4e91-b4c2-000000000003",
    "Timestamp": "2022-10-18T16:03:00Z",
    "RoleIds": [
      "0ae4be37-7156-48a6-8598-000000000003",
      "9f825216-3ed1-4543-89d6-000000000003"
    ]
  },
  {
    "UserId": "51b78930-5854-4e91-b4c2-000000000001",
    "Timestamp": "2022-10-18T16:01:00Z",
    "RoleIds": [
      "0ae4be37-7156-48a6-8598-000000000001",
      "9f825216-3ed1-4543-89d6-000000000001"
    ]
  }
]
